### PR TITLE
Add `workspaces` to `PackageJson` type

### PR DESCRIPTION
**Pull request by ifiokjr**
### Description

Yarn workspaces are becoming a more prominent feature of the `package.json` file.

### CODE REVIEW

File: source/package-json.d.ts [hunk 1]
```diff
@@ -257,8 +257,26 @@ declare namespace PackageJson {
...
+		packages: string[];
+		/**
+		A list of glob patterns for packages that should not be hoisted.
+		*/
+		nohoist: string[];
+	}
+
 	export interface YarnConfiguration {
 		/**
+    Used to configure yarn (and other) workspaces.
```
Comment (sindresorhus) [inline source/package-json.d.ts]: Use tab-indentation.

Would also be useful to link to the Yarn docs about this property.

File: source/package-json.d.ts [hunk 2]
```diff
@@ -257,8 +257,26 @@ declare namespace PackageJson {
...
+
+    Please note private should be set to true to use workspaces
+    */
+    workspaces?: string[] | Workspaces;
```
Comment (sindresorhus) [inline source/package-json.d.ts]: What is `string[]`? I think the docs could be clearer on what it is.

File: source/package-json.d.ts [hunk 3]
```diff
@@ -257,8 +257,26 @@ declare namespace PackageJson {
...
+		A list of glob patterns for packages that should not be hoisted.
+		*/
+		nohoist: string[];
+	}
+
 	export interface YarnConfiguration {
 		/**
+    Used to configure yarn (and other) workspaces.
+
+    Please note private should be set to true to use workspaces
```
Comment (sindresorhus) [inline source/package-json.d.ts]:
```suggestion
    Please note private should be set to true to use workspaces.
```

And what is "private". Would be good to make it clear you're talking about the top-level private property.

File: source/package-json.d.ts (hunk 1 above)
Comment (sindresorhus) [inline source/package-json.d.ts]:
```suggestion
    Used to configure Yarn (and other) workspaces.
```

What "other"?

File: source/package-json.d.ts [hunk 4]
```diff
@@ -257,8 +257,26 @@ declare namespace PackageJson {
 		typings?: string;
 	}
 
+	export interface Workspaces {
+		/**
+		The packages blob to use for workspaces.
```
Comment (sindresorhus) [inline source/package-json.d.ts]: Not clear to me what "packages blob" is.

File: source/package-json.d.ts [hunk 5]
```diff
@@ -257,8 +257,26 @@ declare namespace PackageJson {
...
+		*/
+		packages: string[];
+		/**
+		A list of glob patterns for packages that should not be hoisted.
```
Comment (sindresorhus) [inline source/package-json.d.ts]: I would elaborate on what "hoisting" mean here.

**Comment by sindresorhus** [comment]
It's a welcome addition, but I think the doc comments could be more verbose.

**Comment by ifiokjr** [comment]
I'll address the requested changes soon.

File: source/package-json.d.ts (hunk 5 above)
Comment (ifiokjr) [reply source/package-json.d.ts]: Done, plus I've added a link to the Yarn website.

File: source/package-json.d.ts (hunk 1 above)
Comment (ifiokjr) [reply source/package-json.d.ts]: I've removed the other.

**Comment by ifiokjr** [comment]
I've pushed the requested changes.

**Comment by sindresorhus** [comment]
👌

**Pull request merged**